drodrpg/DROD/GameScreen.cpp

```diff
@@ -2446,9 +2446,13 @@ void CGameScreen::OnClick(
 					if (!this->pCurrentGame->InCombat() && !this->bNeedToProcessDelayedQuestions)
 					{
 						ExploredRoom *pExpRoom = this->pCurrentGame->getExploredRoom(roomID);
-						if (pExpRoom && pExpRoom->HasDetail())
+						if ((pExpRoom && pExpRoom->HasDetail()) ||
+							this->pCurrentGame->pTotalMapStates->GetStoredMapStateForRoom(roomID) >= MapState::Preview)
```

Comment:
Let's avoid calling into members of the CCurrentGame class where possible and keep the codebase simpler by adding a wrapper method in CCurrentGame that queries pTotalMapStates, e.g., in CurrentGame.h:
MapState CCurrentGame::GetStoredMapStateForRoom(const UINT roomID) { return this->pTotalMapStates->GetStoredMapStateForRoom(roomID); }

Then all the callers can call this one.